griptape/artifacts/blob_artifact.py

```diff
@@ -8,7 +8,7 @@
 @define(frozen=True)
 class BlobArtifact(BaseArtifact):
     value: bytes = field(converter=BaseArtifact.value_to_bytes)
-    dir_name: str | None = field(default=None, kw_only=True)
+    dir_name: Optional[str] = field(default=None, kw_only=True)
```

Comment:
Is there any nuance lost between `| None` and `Optional` that could be a gotcha?

Reply:
No gotchas from a day-to-day standpoint. Only when you start doing reflective operations like I'm doing in #561.